spec/models/api_request_spec.rb

```diff
@@ -28,25 +28,64 @@
       end
     end
 
-    describe "without combinations of host, path, and query" do
+    describe "without combinations of host, path, and parameters" do
       it "returns an error for missing host" do
-        # mock in path and query
+        # mock in path and parameters
         allow_any_instance_of(ApiRequest).to receive(:path).and_return "/path"
-        allow_any_instance_of(ApiRequest).to receive(:query).and_return "?some=query"
+        allow_any_instance_of(ApiRequest).to receive(:parameters).and_return []
         expect { ApiRequest.new.response }.to raise_error(RuntimeError)
       end
       it "returns an error for missing path" do
-        # mock in path and query
+        # mock in path and parameters
         allow_any_instance_of(ApiRequest).to receive(:host).and_return "example.com"
-        allow_any_instance_of(ApiRequest).to receive(:query).and_return "?some=query"
+        allow_any_instance_of(ApiRequest).to receive(:parameters).and_return []
         expect { ApiRequest.new.response }.to raise_error(RuntimeError)
       end
       it "returns an error for missing query" do
-        # mock in path and query
+        # mock in path and parameters
         allow_any_instance_of(ApiRequest).to receive(:host).and_return "example.com"
         allow_any_instance_of(ApiRequest).to receive(:path).and_return "/path"
         expect { ApiRequest.new.response }.to raise_error(RuntimeError)
       end
     end
+
+    context "with differing pagination requests" do
+      let(:api) { ApiRequest.new }
+
+      before do
+        success = Net::HTTPSuccess.new(1.0, "200", "OK")
+        allow_any_instance_of(Net::HTTP)
+          .to receive(:start)
+          .and_return(success)
+        expect(success)
+          .to receive(:body)
+          .and_return('{"meta":{},"data":[]}')
+
+        # mock in host, path, and query
+        allow_any_instance_of(ApiRequest).to receive(:host).and_return "example.com"
+        allow_any_instance_of(ApiRequest).to receive(:path).and_return "/path"
+        allow_any_instance_of(ApiRequest).to receive(:parameters).and_return []
+      end
+
+      it "returns no offset if no page is set" do
+        api.response
+        uri = api.instance_variable_get(:@uri)
+        expect(uri.to_s).not_to include("offset")
+        expect(uri.to_s).not_to include("limit")
+      end
+      it "returns the correct offset and limit if page is set to 1" do
+        api.instance_variable_set(:@page, 1)
```

Comment:
[thought] having to set this in this way makes me feel even more sure that this should be set in the system Apis rather than in the general ApiRequest class!